src/createTransformer.ts

```diff
@@ -28,9 +28,7 @@ function isStyledFunction(node: ts.Node, identifiers: CustomStyledIdentifiers):
             return true;
         }
 
-        if (node.name.text === 'extend'
-            && isValidComponent(node.expression)) {
-
+        if (isStyledExtendIdentifier(node.name.text, identifiers)) {
```

Comment:
```suggestion
        if (isStyledExtendIdentifier(node.name.text, identifiers)
            && isValidComponent(node.expression)) {
```

we still need to check that it is a valid component.